api/v1alpha1/gateway_types.go

```diff
@@ -318,6 +318,15 @@ type RouteBindingSelector struct {
 	//
 	// +optional
 	RouteSelector metav1.LabelSelector `json:"routeSelector,omitempty" protobuf:"bytes,2,opt,name=routeSelector"`
+	// Resource specifies the kind of resource to select with the above selectors.
+	//
+	// Resource MUST correspond to route resources that are compatible with the
+	// application protocol specified in the Listener's Protocol field.
+	//
+	// Support: Core
+	//
+	// +required
+	Resource metav1.GroupVersionResource `json:"resource" protobuf:"bytes,3,opt,name=resource"`
```

Comment:
Two questions:

* You use the `metav1.GroupVersionResource` type here, whereas in [the object reference types](https://github.com/kubernetes-sigs/service-apis/blob/d87926665740a8f10011b9f09a44ab3dc4e09403/api/v1alpha1/local_object_reference_types.go), we define explicit `group` and `resource` fields of `string` type, per [upstream guidance](https://github.com/kubernetes/api/blob/6e59bf7a52c237aadf4cc1f905c66b5f178c30b9/core/v1/types.go#L4985-L4996).  Is the use of `metav1.GroupVersionResource` based on any API conventions or upstream guidance?  Would it make sense to use `metav1.GroupVersionResource` in object reference types, or would it perhaps be better to define explicit fields here to be consistent with the object reference types?

* It looks like the above field results in a CRD with fields `spec.listeners[*].routes.resource.group`, `spec.listeners[*].routes.resource.resource`, and `spec.listeners[*].routes.resource.version`.  If we do go with `metav1.GroupVersionResource`, would it make sense to inline it to avoid stutter (`resource.resource`) and reduce the level of nesting?

Reply:
Ah, sorry, Bowei already suggested inlining in https://github.com/kubernetes-sigs/service-apis/pull/226#discussion_r446286586, so it looks like that has already been addressed.

Reply:
I was actually just wrapping up with converting to explicit `resource` and `group` fields of type `string` as you wrote that - it's more consistent with our other uses, upstream guidance, and allows documentation and defaulting for each field which we would lose with inlined GVR. @howardjohn group now defaults to `networking.x-k8s.io`

@bowei @jpeach PTAL